RefsViews/Branches/RefItem.cpp

```diff
@@ -140,28 +149,27 @@ QVariant RefBranch::data(int role) const
 
     switch (role) {
     case Qt::FontRole:
-        #if 0
-        if( mRef.isCurrentBranch() )
+        if( isCurrentBranch( mObject->repository()->HEAD() ) )
         {
             QFont f;
             f.setBold( true );
             return f;
         }
-        #endif
         break;
 
     case RefItem::RowBgGradientRole:
-        #if 0
-        if ( mRef.compare( mRef.repository().HEAD(r) ) == 0 )
-        {
-            QColor back = mRef.isCurrentBranch()
-                          ? QColor::fromHsl(35, 255, 190)
-                          : QColor::fromHsl(35, 255, 190).lighter(130);
-            return back;
+    {
+        RM::Ref* HEAD = mObject->repository()->HEAD();
```

Comment:
Didn't I relatively explicitly say that it is __big bullshit__ to read the repository from within the painting code? I have code that does it correctly, it's not complete and not yet pushed.

Contrary to the RepoMan code (which is prepared to optimise loading / unloading of repositories), this place will always require a loaded repository doing drawing.

Loading a whole repository (just the parts we _need_ here _now_) involves: Reading the configuration, the user configuration, the system configuration and all files included from within there. A recent example if LG PRs showed that this might total to 30 files in a modular setup. It further involves reading all references. Since references can be packed or loose, there's no other way than to read all loose and then amend and obsolete some of them with the packed file. This not only has to physically read the files, but will also have to do lots of memory allocations - just to discard them soon afterwards. A repository may easily have 1000 references! Note that a tag is a reference, too - and each references in a remote is a reference also. So, we already get to 1000 with just 3 remotes and a repository that has release-early-release-often policy, with the 3 typical dev / maint / next branches, 47 topic branches and 200 historic release branches/tags (If release schedule is weekly, this is just 4 years).

Even if all of that is packed into one file, I would dare to say that with the avg. line length in the refs file of 60 bytes, we have to read about 60 Kb of pack file. Then has to be parsed and checked against previous read references (i.e = 2 * n memory allocs and O(n log n) runtime)...

I don't want that to be processed during a mouse move (which in fact may trigger painting or cause the model to be read for other reasons). Other reasons that come in mind: Calculating the item height request knowledge of the font, so a hit-test during mouse move _could_ query for the font, which as we see above reads the repository and then _uses a boolean value_.